pandas/core/resample.py

```diff
@@ -2729,6 +2743,9 @@ def asfreq(
         if how is None:
             how = "E"
 
+        if isinstance(freq, BaseOffset):
+            freq = freq_to_period_freqstr(freq.n, freq.name)
```

Comment:
i think if a user tries to use a period-invalid freq with a PeriodIndex we would want to raise/warn here wouldn't we?

Reply:
it already raises doesn't it?
```python
In [10]: df = pd.DataFrame({'ts': pd.period_range('2000', periods=3), 'val': [1,2,3]})

In [11]: df.set_index('ts').resample('MS').mean()
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[11], line 1
----> 1 df.set_index('ts').resample('MS').mean()

File ~/tmp/.venv/lib/python3.10/site-packages/pandas/core/generic.py:9439, in NDFrame.resample(self, rule, axis, closed, label, convention, kind, on, level, origin, offset, group_keys)
   9436 else:
   9437     axis = 0
-> 9439 return get_resampler(
   9440     cast("Series | DataFrame", self),
   9441     freq=rule,
   9442     label=label,
   9443     closed=closed,
   9444     axis=axis,
   9445     kind=kind,
   9446     convention=convention,
   9447     key=on,
   9448     level=level,
   9449     origin=origin,
   9450     offset=offset,
   9451     group_keys=group_keys,
   9452 )

File ~/tmp/.venv/lib/python3.10/site-packages/pandas/core/resample.py:1970, in get_resampler(obj, kind, **kwds)
   1966 """
   1967 Create a TimeGrouper and return our resampler.
   1968 """
   1969 tg = TimeGrouper(**kwds)
-> 1970 return tg._get_resampler(obj, kind=kind)

File ~/tmp/.venv/lib/python3.10/site-packages/pandas/core/resample.py:2143, in TimeGrouper._get_resampler(self, obj, kind)
   2134     return DatetimeIndexResampler(
   2135         obj,
   2136         timegrouper=self,
   (...)
   2140         gpr_index=ax,
   2141     )
   2142 elif isinstance(ax, PeriodIndex) or kind == "period":
-> 2143     return PeriodIndexResampler(
   2144         obj,
   2145         timegrouper=self,
   2146         kind=kind,
   2147         axis=self.axis,
   2148         group_keys=self.group_keys,
   2149         gpr_index=ax,
   2150     )
   2151 elif isinstance(ax, TimedeltaIndex):
   2152     return TimedeltaIndexResampler(
   2153         obj,
   2154         timegrouper=self,
   (...)
   2157         gpr_index=ax,
   2158     )

File ~/tmp/.venv/lib/python3.10/site-packages/pandas/core/resample.py:178, in Resampler.__init__(self, obj, timegrouper, axis, kind, gpr_index, group_keys, selection)
    173 self.as_index = True
    175 self.obj, self.ax, self._indexer = self._timegrouper._set_grouper(
    176     self._convert_obj(obj), sort=True, gpr_index=gpr_index
    177 )
--> 178 self.binner, self.grouper = self._get_binner()
    179 self._selection = selection
    180 if self._timegrouper.key is not None:

File ~/tmp/.venv/lib/python3.10/site-packages/pandas/core/resample.py:240, in Resampler._get_binner(self)
    234 @final
    235 def _get_binner(self):
    236     """
    237     Create the BinGrouper, assume that self.set_grouper(obj)
    238     has already been called.
    239     """
--> 240     binner, bins, binlabels = self._get_binner_for_time()
    241     assert len(bins) == len(binlabels)
    242     bin_grouper = BinGrouper(bins, binlabels, indexer=self._indexer)

File ~/tmp/.venv/lib/python3.10/site-packages/pandas/core/resample.py:1828, in PeriodIndexResampler._get_binner_for_time(self)
   1826 if self.kind == "timestamp":
   1827     return super()._get_binner_for_time()
-> 1828 return self._timegrouper._get_period_bins(self.ax)

File ~/tmp/.venv/lib/python3.10/site-packages/pandas/core/resample.py:2336, in TimeGrouper._get_period_bins(self, ax)
   2330 if not isinstance(ax, PeriodIndex):
   2331     raise TypeError(
   2332         "axis must be a PeriodIndex, but got "
   2333         f"an instance of {type(ax).__name__}"
   2334     )
-> 2336 memb = ax.asfreq(self.freq, how=self.convention)
   2338 # NaT handling as in pandas._lib.lib.generate_bins_dt64()
   2339 nat_count = 0

File ~/tmp/.venv/lib/python3.10/site-packages/pandas/core/indexes/period.py:182, in PeriodIndex.asfreq(self, freq, how)
    175 @doc(
    176     PeriodArray.asfreq,
    177     other="pandas.arrays.PeriodArray",
   (...)
    180 )
    181 def asfreq(self, freq=None, how: str = "E") -> Self:
--> 182     arr = self._data.asfreq(freq, how)
    183     return type(self)._simple_new(arr, name=self.name)

File ~/tmp/.venv/lib/python3.10/site-packages/pandas/core/arrays/period.py:724, in PeriodArray.asfreq(self, freq, how)
    721 freq = Period._maybe_convert_freq(freq)
    723 base1 = self._dtype._dtype_code
--> 724 base2 = freq._period_dtype_code
    726 asi8 = self.asi8
    727 # self.freq.n can't be negative or 0

AttributeError: 'pandas._libs.tslibs.offsets.MonthBegin' object has no attribute '_period_dtype_code'
```